kythe/cxx/verifier/verifier.cc

```diff
@@ -375,7 +380,9 @@ class Solver {
       }
     } else if (Range* sr = s->AsRange()) {
       if (Range* tr = t->AsRange()) {
-        if (sr->begin() == tr->begin() && sr->end() == tr->end()) {
+        if (sr->begin() == tr->begin() && sr->end() == tr->end() &&
+            sr->path() == tr->path() && sr->root() == tr->root() &&
+            sr->corpus() == tr->corpus()) {
```

Comment:
There are enough components being compared here that this should likely be a separate function or an intrinsic part of Range